### PR TITLE
CI: Add conventional commit checker

### PR DESCRIPTION
**Pull request by ObstinateM**
**Contributor**: Mathis Beauville

**Description:** This pull request adds a new job to the existing GitHub Actions workflow to ensure that all commits follow the "Conventional Commit" format. The commit types are dynamically loaded from a `commit_type.json` file, allowing for easy customization and future scalability.

**Main Changes:**

- Added a new job to the existing GitHub Actions workflow to check commit messages.
- Integrated a custom python script to check commit message.
- [Commit types](https://github.com/commitizen/conventional-commit-types/blob/master/index.json) are loaded from a `commit_type.json` file, making it easier to update and manage allowed types.

**Linked Issue:** #6

**Testing:**

The new job was tested to ensure proper validation of commit messages and correct loading of commit types from commit_type.json.
If a commit doesn't comply with the specified format, the workflow will fail, and an error message will be returned. (see `test/failing` branch)

**Instructions:**

After this PR is merged, contributors must ensure that their commits follow the "Conventional Commit" format, with types as defined in `commit_type.json`. Non-compliant commits will result in PR failure during validation.

### CODE REVIEW

**Comment by SMoraisDev** [comment]
On échangera sur cette PR pendant la prochaine séance, ne touche plus a rien stp :)

**Review by SebastienMorais — CHANGES_REQUESTED** [review]
Peux-tu utiliser des versions d'action plus récentes ?
Soucis de conflit :)

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -11,6 +11,23 @@ on:
...
 
+  lint_commits:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2  # Fetch at least 2 commits to ensure we get non-merge commits
+      - name: Set up Python
+        uses: actions/setup-python@v4
```
[the remark this answers is not in the document]
Comment (ObstinateM) [reply .github/workflows/ci_cd.yml:22]: Les autres actions sont en version 4 😢

[the remark this answers is not in the document]
Comment (SMoraisAnsys) [reply .github/workflows/ci_cd.yml]: On peut discuter de ce point demain si tu veux

[the remark this answers is not in the document]
Comment (ObstinateM) [reply .github/workflows/ci_cd.yml]: Done 🥲

**Pull request merged**